docs/root/intro/arch_overview/security/rbac_filter.rst

```diff
@@ -89,8 +89,21 @@ The following attributes are exposed to the language runtime:
    connection.mtls, bool, Indicates whether TLS is applied to the downstream connection and the peer ceritificate is presented
    connection.requested_server_name, string, Requested server name in the downstream TLS connection
    connection.tls_version, string, TLS version of the downstream TLS connection
+   connection.subject_local_certificate, string, The subject field of the local certificate in the downstream TLS connection
+   connection.subject_peer_certificate, string, The subject field of the peer certificate in the downstream TLS connection
+   connection.dns_san_local_certificate, string, The first DNS entry in the SAN field of the local certificate in the downstream TLS connection
+   connection.dns_san_peer_certificate, string, The first DNS entry in the SAN field of the peer certificate in the downstream TLS connection
+   connection.uri_san_local_certificate, string, The first URI entry in the SAN field of the local certificate in the downstream TLS connection
+   connection.uri_san_peer_certificate, string, The first URI entry in the SAN field of the peer certificate in the downstream TLS connection
    upstream.address, string, Upstream connection remote address
    upstream.port, int, Upstream connection remote port
+   upstream.tls_version, string, TLS version of the upstream TLS connection
+   upstream.subject_local_certificate, string, The subject field of the local certificate in the upstream TLS connection
+   upstream.subject_peer_certificate, string, The subject field of the peer certificate in the upstream TLS connection
+   upstream.dns_san_local_certificate, string, The first DNS entry in the SAN field of the local certificate in the upstream TLS connection
```

Comment:
Shall we return a list instead of just the first one?

Reply:
RBAC, ext authz, and istio use only the first one. Lists would require allocating an extra wrapper unfortunately, so let's not do that until someone asks.